src/main/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFs.java

```diff
@@ -169,36 +187,79 @@ public void apply(VariantContext vc, ReadsContext reads, ReferenceContext ref, F
-                regenotypedVC = GATKVariantContextUtils.reverseTrimAlleles(regenotypedVC);
-                result = addGenotypingAnnotations(originalVC.getAttributes(), regenotypedVC);
+                final VariantContext allelesTrimmed = GATKVariantContextUtils.reverseTrimAlleles(regenotypedVC);
+
+                final VariantContext withAnnotations = addGenotypingAnnotations(originalVC.getAttributes(), allelesTrimmed);
+
+                //TODO: remove this when proper support for reducible annotations is added
+                result = RMS_MAPPING_QUALITY.finalizeRawMQ(withAnnotations);
             }
+        } else {
+            result = originalVC;
         }
 
+
         // if it turned monomorphic then we either need to ignore or fix such sites
         // Note that the order of these actions matters and is different for polymorphic and monomorphic sites.
+        // For polymorphic sites we need to make sure e.g. the SB tag is sent to the annotation engine and then removed later.
+        // For monomorphic sites we need to make sure e.g. the hom ref genotypes are created and only then are passed to the annotation engine.
+        // We could theoretically make 2 passes to re-create the genotypes, but that gets extremely expensive with large sample sizes.
         if ( result.isMonomorphicInSamples() ) {
             if ( !includeNonVariants) {
                 return null;
             } else {
                 // For monomorphic sites we need to make sure e.g. the hom ref genotypes are created and only then are passed to the annotation engine.
                 final VariantContext reannotated = new VariantContextBuilder(result).genotypes(cleanupGenotypeAnnotations(result, true)).make();
-                return annotationEngine.annotateContext(reannotated, features, ref, null, a -> true);
+                return annotationEngine.annotateContext(reannotated, features, ref, null, GenotypeGVCFs::annotationShouldBeSkippedForHomRefSites);
             }
         } else {
             // For polymorphic sites we need to make sure e.g. the SB tag is sent to the annotation engine and then removed later.
             final VariantContext reannotated = annotationEngine.annotateContext(result, features, ref, null, a -> true);
             return new VariantContextBuilder(reannotated).genotypes(cleanupGenotypeAnnotations(reannotated, false)).make();
         }
+    }
+
+    private VariantContext calculateGenotypes(VariantContext vc){
+
+        final VariantContext.Type type = vc.getType();
+        final GenotypeLikelihoodsCalculationModel model;
+        /*
+         * Query the VariantContext for the appropriate model.  If type == MIXED, one would want to use model = BOTH.
+         * However GenotypingEngine.getAlleleFrequencyPriors throws an exception if you give it anything but a SNP or INDEL model.
+         */
+        if ( type == VariantContext.Type.INDEL ) {
+            model = GenotypeLikelihoodsCalculationModel.INDEL;
```

Comment:
We definitely have indels in our test inputs.  I think they're representative, but maybe not exhaustive. As far as I can tell, the only thing the model is used for in genotypeGVCFs is in that it's passed through to `getAlleleFrequencyPriors` which is a method on `GenotypingEngine` that was ported to gatk4. As far as I can see everythign we need is here.  The model seems to be used in a number of methods on UnifiedGenotypingEngine that we DON'T call from GenotypeGVCFs.